TOL.format() fail for integer input under IronPython environment

**Describe the bug**
I was running through some compas_fab test and i noticed an error related to TOL.format(), it only happens in the IPY environment.

The failed build error message:

https://github.com/compas-dev/compas_fab/actions/runs/7751943548/job/21140627701?pr=400

```
==================================== FAILURES ====================================
 tests\robots\test_robot.py::test_inverse_kinematics_repeated_calls_will_return_next_result[1]
Traceback (most recent call last):
  File "C:\ProgramData\chocolatey\lib\ironpython\Lib\site-packages\pytest\test_runner.py", line 137, in run
    test_method(**kwargs)
  File "tests\robots\test_robot.py", line 349, in test_inverse_kinematics_repeated_calls_will_return_next_result
    configuration = robot.inverse_kinematics(frame, start_config)
  File "c:\users\leungp\documents\github\compas_fab\src\compas_fab\robots\robot.py", line 1305, in inverse_kinematics
    request_id = "{}-{}-{}-{}-{}".format(
  File "c:\users\leungp\documents\github\compas_robots\src\compas_robots\configuration.py", line 232, in __str__
    values_string = ", ".join((TOL.format_number(v) for v in self.joint_values))
  File "c:\users\leungp\documents\github\compas_robots\src\compas_robots\configuration.py", line 232, in <genexpr>
    values_string = ", ".join((TOL.format_number(v) for v in self.joint_values))
  File "c:\users\leungp\documents\github\compas_2\src\compas\tolerance.py", line 700, in format_number
    return "{0:.{1}f}".format(number, precision)
ValueError: Precision not allowed in integer format specifier

============================ short test summary info ============================
FAILED tests\robots\test_robot.py::test_inverse_kinematics_repeated_calls_will_return_next_result[1] - Precision not allowed in integer format specifier
========================= 1 failed, 81 passed in 14.82s =========================
```
I traced it down to the function where in Python 2, we simply cannot pass integers into those formatters with precision defined.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+* Fixed bug in `compas.tolerange.Tolerance.format_number()` related to IronPython environment.
+
 ### Removed
 
 

--- a/src/compas/tolerance.py
+++ b/src/compas/tolerance.py
@@ -9,6 +9,7 @@ from __future__ import division
 from compas.data import Data
 from decimal import Decimal
 
+import compas
 
 __all__ = ["Tolerance", "TOL"]
 
@@ -683,6 +684,9 @@ class Tolerance(Data):
         '12300'
 
         """
+        if compas.IPY:
+            number = float(number)
+
         if not precision:
             precision = self.precision
 

